tests/unit/synapseutils/unit_test_synapseutils_sync.py

```diff
@@ -773,3 +777,160 @@ def test_get_file_entity_provenance_dict__error_not_404(self):
+    path1 = os.path.abspath(os.path.expanduser('~/file1.txt'))
+    path2 = 'http://www.synapse.org'
+    path3 = os.path.abspath(os.path.expanduser('~/file3.txt'))
+    path4 = 'http://www.github.com'
+
+    row1 = f'{path1}\t{project_id}\n'
+    row2 = f'{path2}\t{project_id}\n'
+    row3 = f'{path3}\t{project_id}\n'
+    row4 = f'{path4}\t{project_id}\n'
+
+    mock_os.reset_mock()
+    manifest = StringIO(header + row1 + row2 + row3 + row4)
+    mock_os.path.isfile.side_effect = [True, True, True, False]
+    mock_os.path.abspath.side_effect = [path1, path3]
+    mock_stat = MagicMock(spec='st_size')
+    mock_os.stat.return_value = mock_stat
+    mock_stat.st_size = 0
+    with pytest.raises(ValueError) as ve:
+        sync.readManifestFile(syn, manifest)
+    assert str(ve.value) == "All the files uploaded cannot be 0 byte."
+
+
+@patch.object(sync, 'os')
+def test_check_file_name(mock_os, syn):
+    """
+    Verify the check_file_name method works correctly
+    """
+
+    project_id = "syn123"
+    header = 'path\tparent\tname\n'
+    path1 = os.path.abspath(os.path.expanduser('~/file1.txt'))
+    path2 = os.path.abspath(os.path.expanduser('~/file2.txt'))
+    path3 = os.path.abspath(os.path.expanduser('~/file3.txt'))
+
+    row1 = f"{path1}\t{project_id}\tTest_file_name.txt\n"
+    row2 = f"{path2}\t{project_id}\tTest_file-name`s(1).txt\n"
+    row3 = f"{path3}\t{project_id}\t\n"
+
+    manifest = StringIO(header + row1 + row2 + row3)
+    mock_os.path.isfile.side_effect = [True, True, True]
+    mock_os.path.abspath.side_effect = [path1, path2, path3]
+    mock_os.path.basename.return_value = 'file3.txt'
+
+    # mock syn.get() to return a project because the final check is making sure parent is a container
+    # mock isfile() to always return true to avoid having to create files in the home directory
+    # side effect mocks values for: manfiest file, file1.txt, file2.txt, isfile(project.id) check in syn.get()
+    with patch.object(syn, "get", return_value=Project()):
+        sync.readManifestFile(syn, manifest)
+
+
+@patch.object(sync, 'os')
+def test_check_file_name_with_illegal_char(mock_os, syn):
+    """
+    Verify the check_file_name method raises the ValueError when the file name contains illegal char
+    """
+
+    project_id = "syn123"
+    header = 'path\tparent\tname\n'
+    path1 = os.path.abspath(os.path.expanduser('~/file1.txt'))
+    path2 = os.path.abspath(os.path.expanduser('~/file2.txt'))
+    path3 = os.path.abspath(os.path.expanduser('~/file3.txt'))
+    path4 = os.path.abspath(os.path.expanduser('~/file4.txt'))
+
+    row1 = f"{path1}\t{project_id}\tTest_file_name.txt\n"
+    row2 = f"{path2}\t{project_id}\tTest_file-name`s(1).txt\n"
+    row3 = f"{path3}\t{project_id}\t\n"
+    row4 = f"{path4}\t{project_id}\tTest_file_name_with_#.txt\n"
+
+    manifest = StringIO(header + row1 + row2 + row3 + row4)
+    mock_os.reset_mock()
+    mock_os.path.isfile.side_effect = [True, True, True, True]
```

Comment:
Since the return value of this never changes could it be isfile.return_value = True rather than using a side_effect?